File: app/src/main/java/zone/com/zanimate/ViewWrap.java
package zone.com.zanimate;
import android.view.View;

import zone.com.zanimate.value.ValueAnimatorProxy;
import zone.com.zanimate.value.ValueAnimatorProxyAbstract;

/**
 * Created by fuzhipeng on 16/8/23.
 */
public class ViewWrap extends ValueAnimatorProxyAbstract<ViewWrap> {
    private View view;

    public ViewWrap() {
        child = this;//一定要设置 没设置到时候会抛异常给你;
    }

    @Override
    public android.animation.ValueAnimator initDefaultValueAnimator() {
        return ValueAnimatorProxy.ofFloat(0F, 1F)
//                .setRepeatCount(ValueAnimatorProxy.INFINITE)
                .setRepeatMode(android.animation.ValueAnimator.REVERSE)
                .setDuration(1000).source();
    }

    public void setShineView(View view) {
        this.view = view;
    }

    @Override
    public ViewWrap start() {
        addUpdateListener(new android.animation.ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(android.animation.ValueAnimator animation) {
                System.out.println("ViewWrap测试~继承好使不~"+animation.getAnimatedValue());
            }
        });
        return super.start();
    }

    @Override
    public ViewWrap addUpdateListener(android.animation.ValueAnimator.AnimatorUpdateListener listener) {
        return super.addUpdateListener(listener);
    }
}

File: app/src/main/java/zone/com/zanimate/ViewWrap.kt
package zone.com.zanimate

import android.view.View

import zone.com.zanimate.value.ValueAnimatorProxy
import zone.com.zanimate.value.ValueAnimatorProxyAbstract

/**
 * Created by fuzhipeng on 16/8/23.
 */
class ViewWrap : ValueAnimatorProxyAbstract<ViewWrap>() {
    private var view: View? = null

    init {
        child = this//一定要设置 没设置到时候会抛异常给你;
    }

    override fun initDefaultValueAnimator(): android.animation.ValueAnimator {
        return ValueAnimatorProxy.ofFloat(0f, 1f)
                //                .setRepeatCount(ValueAnimatorProxy.INFINITE)
                .setRepeatMode(android.animation.ValueAnimator.REVERSE)
                .setDuration(1000).source()
    }

    fun setShineView(view: View) {
        this.view = view
    }

    override fun start(): ViewWrap {
        addUpdateListener { animation -> println("ViewWrap测试~继承好使不~" + animation.animatedValue) }
        return super.start()
    }

    override fun addUpdateListener(listener: android.animation.ValueAnimator.AnimatorUpdateListener): ViewWrap {
        return super.addUpdateListener(listener)
    }
}
